exporter/collector/logs.go

```diff
@@ -43,20 +74,107 @@ func (l *LogsExporter) Shutdown(ctx context.Context) error {
 }
 
 func (l *LogsExporter) PushLogs(ctx context.Context, ld pdata.Logs) error {
+	mapper := &metricMapper{} // Refactor metricMapper to map MRs for logging?
+	logPushErrors := []error{}
 	for i := 0; i < ld.ResourceLogs().Len(); i++ {
 		rl := ld.ResourceLogs().At(i)
-		mapper := &metricMapper{} // Refactor metricMapper to map MRs for logging?
 		mr, _ := mapper.resourceToMonitoredResource(rl.Resource())
 		for j := 0; j < rl.InstrumentationLibraryLogs().Len(); j++ {
 			ill := rl.InstrumentationLibraryLogs().At(j)
 			for k := 0; k < ill.LogRecords().Len(); k++ {
 				log := ill.LogRecords().At(k)
-				l.logger.Log(logging.Entry{
-					Resource: mr,
-					Payload:  log.Body().AsString(),
-				})
+
+				entry, err := l.mapper.logToEntry(log, mr)
+				if err != nil {
+					logPushErrors = append(logPushErrors, err)
+				} else {
+					l.logger.Log(entry)
+				}
 			}
 		}
 	}
+	if len(logPushErrors) > 0 {
+		return multierr.Combine(logPushErrors...)
+	}
 	return nil
 }
+
+func (l logMapper) logToEntry(
+	log pdata.LogRecord,
+	mr *monitoredres.MonitoredResource,
+) (logging.Entry, error) {
+	entry := logging.Entry{
+		Resource: mr,
+	}
+
+	logBody := log.Body().BytesVal()
+	if len(logBody) > 0 {
+		entry.Payload = json.RawMessage(logBody)
+	}
+
+	httpRequestAttr, ok := log.Attributes().Get(HTTPRequestAttributeKey)
+	if ok {
+		httpRequest, err := l.parseHTTPRequest(httpRequestAttr.BytesVal())
```

Comment:
I actually don't know if this is going to be bytes or string in a typical use case. @damemi if you happen to know offhand let me know, otherwise I'll manually test it tomorrow.